django/middleware/csrf.py

```diff
@@ -221,7 +222,41 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 # branches that call reject().
                 return self._accept(request)
 
-            if request.is_secure():
+            # Verify the Origin header, if present.
+            origin_verified = False
+            if 'HTTP_ORIGIN' in request.META:
+                good_origin = '%s://%s' % (
+                    'https' if request.is_secure() else 'http',
+                    request.get_host(),
+                )
+                allowed_origins = [good_origin]
+                # Tuples of (allowed scheme, allowed netloc), where all
+                # subdomains of netloc are allowed.
```

Comment:
Is this accurate? From the looks of it, subdomains are only allowed if a '*' is in there.

Reply:
Yes, only `CSRF_TRUSTED_ORIGINS` values with a star are put in `allowed_origin_subdomains`. Would it be more clear if the variable name were `allowed_origin_all_subdomains` or something?

Reply:
Nope, I guess it is as good as it gets. One option would be to renamed `allowed_origins` to `allowed_origins_exact` but not sure if this is worth it.